clang/lib/CIR/Lowering/DirectToLLVM/LowerToLLVM.cpp

```diff
@@ -851,8 +851,9 @@ mlir::LogicalResult CIRToLLVMPtrStrideOpLowering::matchAndRewrite(
     }
   }
 
-  rewriter.replaceOpWithNewOp<mlir::LLVM::GEPOp>(
-      ptrStrideOp, resultTy, elementTy, adaptor.getBase(), index);
+  rewriter.replaceOpWithNewOp<mlir::LLVM::GEPOp>(ptrStrideOp, resultTy,
+                                                 elementTy, adaptor.getBase(),
+                                                 index, /*inbounds=*/true);
```

Comment:
It's still unconditional. We cannot accept miscompilation. Please make sure we emit the same LLVM IR as e.g. [this godbolt](https://godbolt.org/z/rKPKva5ab), or optionally lead it to an assersion failure.

Reply:
ok, I will work on it later.